groupcache.go

```diff
@@ -26,6 +26,7 @@ limitations under the License.
 // Put supports manual loading of a data's canonical owner's cache.
 package groupcache
 
+//TODO(apratti): need to fix mem leak
```

Comment:
This is in reference to the fact that when a HEAD/GET request for a resource that does not exist is made and it queries the metadata cache it will create a cacheValueMetadata that remains indefinitely.

Reply:
We do we need to add the entry in the contain case?

Reply:
Only if we want to cache Contain checks.

Reply:
I just fixed the memory leak for Get, if we remove cacheing for Contain check, we are all clear.

Reply:
Yeah - prefer to not cache contain checks for now